File: apps/hello-world-express/src/dtos/event.ts
import { DomainEvent, User } from '@abgov/adsp-service-sdk';

export const HelloWorldEventName = 'hello-world-event';
export const createHelloEvent = (
  from: User,
  message: string,
  response: string
): DomainEvent => ({
  name: HelloWorldEventName,
  timestamp: new Date(),
  payload: {
    fromUserId: from.id,
    fromUser: from.name,
    message,
    response
  },
});

File: apps/hello-world-django/hello_world_django/event.py
from datetime import datetime
from adsp_service_flask_sdk import DomainEvent, User


HELLO_WORLD_EVENT = "hello-world-event"


def create_hello_event(user: User, message: str, response: str):
    return DomainEvent(
        HELLO_WORLD_EVENT,
        datetime.utcnow(),
        {
            "fromUserId": user.id,
            "fromUser": user.name,
            "message": message,
            "response": response,
        },
    )
